src/v/cluster/archival_metadata_stm.cc

```diff
@@ -91,22 +117,18 @@ archival_metadata_stm::segments_from_manifest(
           .meta = meta});
     }
 
-    std::sort(
-      segments.begin(), segments.end(), [](const auto& s1, const auto& s2) {
-          return s1.meta.base_offset < s2.meta.base_offset;
-      });
-
     return segments;
 }
 
 ss::future<> archival_metadata_stm::make_snapshot(
   const storage::ntp_config& ntp_cfg,
   const cloud_storage::partition_manifest& m,
-  model::offset insync_offset) {
+  model::offset insync_offset,
+  model::offset start_offset) {
     // Create archival_stm_snapshot
     auto segments = segments_from_manifest(m);
     iobuf snap_data = serde::to_iobuf(
-      snapshot{.segments = std::move(segments)});
+      snapshot{.segments = std::move(segments), .start_offset = start_offset});
```

Comment:
I think this works, but the snapshot format should probably record segments + replaced separately now -- so that we aren't doing the work of figuring out overlaps each time we load the snapshot.

Probably something to improve in future.

Reply:
it's already changed. the snapshot contains two separate collections and offsets